app/upgrades/v3/constants.go

```diff
@@ -75,20 +75,17 @@ var ICAAllowMessages = []string{
 	sdk.MsgTypeURL(&govv1beta1.MsgSubmitProposal{}),
 	sdk.MsgTypeURL(&govv1beta1.MsgDeposit{}),
 	sdk.MsgTypeURL(&govv1beta1.MsgVote{}),
-	// Change: add messages from Oracle module
-	sdk.MsgTypeURL(&oracletypes.MsgActivate{}),
-	sdk.MsgTypeURL(&oracletypes.MsgCreateDataSource{}),
-	sdk.MsgTypeURL(&oracletypes.MsgCreateOracleScript{}),
-	sdk.MsgTypeURL(&oracletypes.MsgEditDataSource{}),
-	sdk.MsgTypeURL(&oracletypes.MsgEditOracleScript{}),
-	sdk.MsgTypeURL(&oracletypes.MsgReportData{}),
-	sdk.MsgTypeURL(&oracletypes.MsgRequestData{}),
-
 	sdk.MsgTypeURL(&stakingtypes.MsgEditValidator{}),
 	sdk.MsgTypeURL(&stakingtypes.MsgDelegate{}),
 	sdk.MsgTypeURL(&stakingtypes.MsgUndelegate{}),
 	sdk.MsgTypeURL(&stakingtypes.MsgBeginRedelegate{}),
 	sdk.MsgTypeURL(&stakingtypes.MsgCreateValidator{}),
 	sdk.MsgTypeURL(&vestingtypes.MsgCreateVestingAccount{}),
 	sdk.MsgTypeURL(&ibctransfertypes.MsgTransfer{}),
+	// Change: add messages from Oracle module
+	sdk.MsgTypeURL(&oracletypes.MsgCreateDataSource{}),
+	sdk.MsgTypeURL(&oracletypes.MsgCreateOracleScript{}),
+	sdk.MsgTypeURL(&oracletypes.MsgEditDataSource{}),
+	sdk.MsgTypeURL(&oracletypes.MsgEditOracleScript{}),
+	sdk.MsgTypeURL(&oracletypes.MsgRequestData{}),
```

Comment:
why remove msgactivate and msgreportdata

Reply:
it's the msgs for messages. we should not allow to do it thourgh ica.